src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -386,84 +399,188 @@ private Coordinate getRandomCoordinate(){
         return new Coordinate(randRow,randCol);
 
     }
-
-
-    public void shootAtPlayer() {
+    public void smartShootAtPlayer() {
```

Comment:
We can talk about it tomorrow but, this function is huge. I realize its the most complicated piece of code in our program but maybe we should find a way to break it up? Lets look through this thing tomorrow and figure it out.